src/hotspot/share/cds/classListParser.cpp

```diff
@@ -85,10 +85,12 @@ bool ClassListParser::is_parsing_thread() {
 }
 
 ClassListParser::~ClassListParser() {
-  if (_file) {
+  if (_file != NULL) {
     fclose(_file);
   }
   Atomic::store(&_parsing_thread, (Thread*)NULL);
+  delete _indy_items;
+  delete _interfaces;
```

Comment:
In fact below _instance = NULL is not necessary. ClassListParser is a stack object and in ctor, it sets 
_instance = this;
So after ClassListParser deallocated from stack, setting it to NULL makes no sense.
If the it is heap object, setting it to NULL could not prevent memory leak.
Should we remove it?

Reply:
You can ignore my comment here, It is still useful to set it to NULL:
  static ClassListParser* instance() {
    assert(is_parsing_thread(), "call this only in the thread that created ClassListParsing::_instance");
    assert(_instance != NULL, "must be");
    return _instance;
  }
The static function will check if it is NULL.